### ISSUE TITLE
Possible false negative matches for properties with union types

### ISSUE THREAD

**Issue by lucas-koehler** [eclipse-theia/vscode-theia-comparator#26]
The `TreeItem.label` property is an optional property of type `string | TreeItemLabel` in the vscode API. In Theia's API it currently is an optional property with only the type `string` (see [this permalink](https://github.com/eclipse-theia/theia/blob/b2e44a9367b1d9c9b1a0977edf57deab3cc50759/packages/plugin/src/theia.d.ts#L5337)). Nevertheless, the `label` property is shown as implemented in the report.

Maybe this is related to the type `TreeItemLabel` not being present in Theia's API def. I could imagine this resolves the type as `any` and thus matches every time.

**Comment by colin-grant-work** [comment]
I am seeing a similar problem with the `languages.createLanguageStatusItem` function, which was implemented for Theia@1.26.0, and has the following declaration, identical in theia.d.ts and vscode.d.ts:

```typescript
/**
		 * Creates a new {@link LanguageStatusItem language status item}.
		 *
		 * @param id The identifier of the item.
		 * @param selector The document selector that defines for what editors the item shows.
		 */
		export function createLanguageStatusItem(id: string, selector: DocumentSelector): LanguageStatusItem;
```

It appears that when parsing `vscode.d.ts`, the type of the `selector` argument resolved to `any` while when parsing `theia.d.ts` it resolves to `DocumentSelector`, and since those don't match, the function is declared not to match.

**Comment by colin-grant-work** [comment]
@lucas-koehler, I believe that this has been addressed by the changes in #36. Running a comparison between Theia@1.26.0 (before the `TreeItem.label` type was brought into conformity) and VSCode@main now shows the `label` field as 'partially' supported:

![image](https://user-images.githubusercontent.com/62660806/192377784-e2a3b24d-b12d-4ee7-893a-59a6e0721f70.png)

**Comment by lucas-koehler** [comment]
@colin-grant-work Thanks for the info. 'Partial' seems to be the correct clarification and looking at your PR it makes sense that it fixes this problem. I'll close this.
